### PR TITLE
Teams Functionality

### PR DESCRIPTION
**Pull request by changjy44**
## Changes

1. Add commands SetTeam, AddTeam and DeleteTeam and respective parsers
2. Modified Addressbook to have multiple team objects.
3. Add storage features to teams upon adding/deleting teams and exiting the program.
4. Add basic tests.

## Things to note.

- I have implemented the addressbook such that the user must be on a current team (i.e. the Team reference in addressbook cannot be null). A default team is always created if the user does not have any teams. 
- Team names must be unique and hence Teams are equal to each other through comparison of their names only (Instead of with their members and tasks).
- The UI is not updated and will be updated after we discuss about it.

### CODE REVIEW

File: src/main/java/seedu/address/model/team/TaskList.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ public int hashCode() {
     @Override
     public String toString() {
         if (internalList.isEmpty()) {
-            return NO_TASKS;
+            return ""; //return NO_TASKS;
```
Comment (marcuspang) [inline src/main/java/seedu/address/model/team/TaskList.java:110]: Why was this changed?

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 2]
```diff
@@ -54,6 +60,29 @@ public AddressBook toModelType() throws IllegalValueException {
...
+        }
+
+        if (counter == 0) {
+            Team team = new Team("default", new ArrayList<>(), new ArrayList<>());
+            addressBook.addTeam(team);
+            addressBook.setTeam(team);
+        }
+
+
+
```
Comment (marcuspang) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:85]:
```suggestion
```

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 3]
```diff
@@ -20,15 +21,19 @@
 class JsonSerializableAddressBook {
 
     public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate person(s).";
-
+    public static final String MESSAGE_DUPLICATE_TEAMS = "Team list contains duplicate team(s).";
     private final List<JsonAdaptedPerson> persons = new ArrayList<>();
```
Comment (marcuspang) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:26]:
```suggestion
```

File: src/main/java/seedu/address/model/Model.java [hunk 4]
```diff
@@ -80,6 +80,18 @@ public interface Model {
     /** Returns the current team */
     Team getTeam();
 
+    /** Sets the current team to an existing team*/
```
Comment (marcuspang) [inline src/main/java/seedu/address/model/Model.java:83]:
```suggestion
    /** Sets the current team to an existing team */
```

File: src/main/java/seedu/address/model/team/exceptions/DuplicateTeamException.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package seedu.address.model.team.exceptions;
+
+/**
+ * Signals that the operation will result in duplicate Teams
```
Comment (marcuspang) [inline src/main/java/seedu/address/model/team/exceptions/DuplicateTeamException.java:4]:
```suggestion
 * Signals that the operation will result in duplicate teams
```

**Review by marcuspang — APPROVED** [review]
Looks good, just some minor nits

File: src/main/java/seedu/address/logic/commands/DeleteTeamCommand.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            throw new CommandException(MESSAGE_TEAM_NOT_EXISTS);
+        }
+
+        if (teamList.size() == 1) {
+            throw new CommandException(MESSAGE_AT_LEAST_ONE_TEAM);
+        }
+
+        model.deleteTeam(teamList.get(teamIndex));
+        if (currentTeam.equals(targetTeam)) {
+            model.setTeam(model.getTeamList().get(0));
```
Comment (marcuspang) [inline src/main/java/seedu/address/logic/commands/DeleteTeamCommand.java:46]: Not too sure if default team is always set to be the 0 index. Instead we can throw an error telling the user that his current team is being deleted

File: src/main/java/seedu/address/model/team/TaskList.java (hunk 1 above)
Comment (changjy44) [reply src/main/java/seedu/address/model/team/TaskList.java:110]: image

I set it to an empty string for now so that the team commands will only show the name of the team, and not other information. Can discuss further in the next meeting.

Comment (marcuspang) [reply src/main/java/seedu/address/model/team/TaskList.java:110]: Okay, can you add a comment above instead of leaving it to the right
